src/main/ruby/truffleruby/core/process.rb

```diff
@@ -403,7 +403,12 @@ def self.euid=(uid)
     # the 4 rescue clauses below are needed
     # until respond_to? can be used to query the implementation of methods attached via FFI
     # atm respond_to returns true if a method is attached but not implemented on the platform
-    uid = Truffle::Type.coerce_to uid, Integer, :to_int
+    if uid.kind_of?(String)
+      uid = Etc.getpwnam(uid).uid
```

Comment:
This needs `require 'etc'` above it. That's in general not ideal but it's a reasonable side for such a rarely-used method, as mentioned in the issue.